v3/project.go

```diff
@@ -23,29 +23,26 @@ func (project) kind() operatorKind {
 
 func (project) layout() exprLayout {
 	return exprLayout{
-		numAux:      2,
+		numAux:      1,
```

Comment:
Why not make an explicit project operator as well? I think making every operator a possible project operator makes matching and replacing messier and more error prone.

Reply:
Project is an explicit operator. It takes as input another relational operator and a list of projections. Only the project operator allows for the `projections` list.

Reply:
OK, I missed that. What is the other remaining "aux" slot used for?

Reply:
`numAux` is the total number of auxiliary slots. Before addition of `select` I needed operators to manually specify that total because `scan` had 1 auxiliary slot for filters but they were at position 0. I think I can remove the need to manually specify `numAux` now. I'll do that in a follow-on PR.